libyara/include/yara/lnk.h

```diff
@@ -0,0 +1,293 @@
+  uint32_t icon_index;
+  uint32_t show_command;
+  uint16_t hotkey_flags;
+  uint16_t reserved1;
+  uint32_t reserved2;
+  uint32_t reserved3;
+} shell_link_header_t;
+
+typedef struct _link_info_fixed_header_t
+{
+  uint32_t link_info_size;
+  uint32_t link_info_header_size;
+  uint32_t link_info_flags;
+  uint32_t volume_id_offset;
+  uint32_t local_base_path_offset;
+  uint32_t common_network_relative_link_offset;
+  uint32_t common_path_suffix_offset;
+} link_info_fixed_header_t;
+
+typedef struct _volume_id_t
+{
+  uint32_t volume_id_size;
+  uint32_t drive_type;
+  uint32_t drive_serial_number;
+  uint32_t volume_label_offset;
+} volume_id_t;
+
+typedef struct _common_network_relative_link_t
+{
+  uint32_t common_network_relative_link_size;
+  uint32_t common_network_relative_link_flags;
+  uint32_t net_name_offset;
+  uint32_t device_name_offset;
+  uint32_t network_provider_type;
+} common_network_relative_link_t;
+
+typedef struct _console_data_block_t
+{
+  uint16_t fill_attributes;
+  uint16_t popup_fill_attributes;
+  uint16_t screen_buffer_size_x;
+  uint16_t screen_buffer_size_y;
+  uint16_t window_size_x;
+  uint16_t window_size_y;
+  uint16_t window_origin_x;
+  uint16_t window_origin_y;
+  uint32_t unused_1;
+  uint32_t unused_2;
+  uint32_t font_size;
+  uint32_t font_family;
+  uint32_t font_weight;
+  uint16_t face_name[32];
+  uint32_t cursor_size;
+  uint32_t full_screen;
+  uint32_t quick_edit;
+  uint32_t insert_mode;
+  uint32_t auto_position;
+  uint32_t history_buffer_size;
+  uint32_t number_of_history_buffers;
+  uint32_t history_no_dup;
+  uint32_t color_table[16];
+} console_data_block_t;
+
+typedef struct _console_fe_data_block_t
+{
+  uint32_t code_page;
+} console_fe_data_block_t;
+
+typedef struct _darwin_data_block_t
+{
+  char darwin_data_ansi[260];
+  uint16_t darwin_data_unicode[260];
+} darwin_data_block_t;
+
+typedef struct _environment_variable_data_block_t
+{
+  char target_ansi[260];
+  uint16_t target_unicode[260];
+} environment_variable_data_block_t;
+
+typedef struct _icon_environment_data_block_t
+{
+  char target_ansi[260];
+  uint16_t target_unicode[260];
+} icon_environment_data_block_t;
+
+typedef struct _known_folder_data_block_t
+{
+  uint8_t known_folder_id[16];
+  uint32_t offset;
+} known_folder_data_block_t;
+
+typedef struct _special_folder_data_block_t
+{
+  uint32_t special_folder_id;
+  uint32_t offset;
+} special_folder_data_block_t;
+
+typedef struct _tracker_data_block_t
+{
+  uint32_t length;
+  uint32_t version;
+  char machine_id[16];
+  uint8_t droid_volume_identifier[16];
+  uint8_t droid_file_identifier[16];
+  uint8_t droid_birth_volume_identifier[16];
+  uint8_t droid_birth_file_identifier[16];
+} tracker_data_block_t;
+
+#pragma pack(pop)
+
+#define HEADER_SIZE  0x0000004C
+#define LINK_CLSID_0 0x00021401
+#define LINK_CLSID_1 0x00000000
+#define LINK_CLSID_2 0x000000C0
+#define LINK_CLSID_3 0x46000000
+
+#define HasLinkTargetIDList         0x00000001
```

Comment:
Some constants use snake case all in uppercases, but these ones use camel-case. Should we use the same naming style for all constants? Are these constants named this way because of something?

Reply:
Good question, I named some of them based off the casing Microsoft used in their docs (e.g. `HasLinkTargetIDList` has that casing in the LNK documentation - https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-SHLLINK/%5bMS-SHLLINK%5d.pdf).

I'm happy to make this consistent throughout however!

Reply:
Updated in https://github.com/VirusTotal/yara/pull/1732/commits/5de9c2c1dd26795398d4db3309cc89e497bcce87